service/matching/matchingEngine_test.go

```diff
@@ -194,11 +194,23 @@ func (s *matchingEngineSuite) TestAckManager() {
 }
 
 func (s *matchingEngineSuite) TestPollForActivityTasksEmptyResult() {
-	s.PollForTasksEmptyResultTest(persistence.TaskListTypeActivity)
+	s.PollForTasksEmptyResultTest(s.callContext, persistence.TaskListTypeActivity)
 }
 
 func (s *matchingEngineSuite) TestPollForDecisionTasksEmptyResult() {
-	s.PollForTasksEmptyResultTest(persistence.TaskListTypeDecision)
+	s.PollForTasksEmptyResultTest(s.callContext, persistence.TaskListTypeDecision)
+}
+
+func (s *matchingEngineSuite) TestPollForActivityTasksEmptyResultWithShortContext() {
+	callContext, cancel := context.WithTimeout(s.callContext, returnEmptyTaskTimeBudget+10*time.Millisecond)
```

Comment:
So right now if somehow matching gets a context with a really short timeout we will still handle the request. But in frontend we say if we ever get a short timeout it's an error case right off the bat. I think it makes sense to also have these super short context timeout checks in matching. 

It may actually make sense to only have the checks exists in matching and make frontend dumber about the context timeout checks?